app/models/post.rb

```diff
@@ -13,7 +13,7 @@ class Post < ApplicationRecord
   belongs_to :last_user, class_name: 'User', inverse_of: false, optional: false
   belongs_to :last_reply, class_name: 'Reply', inverse_of: false, optional: true
   has_one :flat_post, dependent: :destroy
-  has_many :replies, inverse_of: :post, dependent: :delete_all
+  has_many :replies, -> { where('reply_order > 0') }, inverse_of: :post, dependent: :delete_all
```

Comment:
The general strategy here is to make `post.replies` default to ignoring `post.written`, and to update all the `Reply.` queries to explicitly ignore it, right?

Reply:
Yeah. I might have missed a few instances because I started by specifically excluding it everywhere, but...